app/Actions/Photo/Pipes/Shared/HydrateMetadata.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Actions\Photo\Pipes\Shared;
+
+use App\Contracts\PhotoCreate\SharedPipe;
+use App\DTO\PhotoCreate\DuplicateDTO;
+
+/**
+ * Hydrates meta-info of the media file from the
+ * {@link AddStrategyParameters::$exifInfo} attribute of the associated
+ * {@link AddStrategyParameters} object into the associated {@link Photo}
+ * object.
+ *
+ * Meta information is conditionally copied if and only if the target
+ * attribute of the {@link Photo} object is null or empty and the
+ * meta-info is not.
+ * This way this method is usable by {@link AddStandaloneStrategy} and
+ * {@link AddDuplicateStrategy}.
+ * For a freshly created {@link Photo} object (with empty attributes)
+ * all available meta-data is hydrated, but for an already existing
+ * {@link Photo} object existing attributes are not overwritten.
+ */
+class HydrateMetadata implements SharedPipe
+{
+	public function handle(DuplicateDTO $state, \Closure $next): DuplicateDTO
+	{
+		if ($state->photo->title === null) {
+			$state->photo->title = $state->exifInfo->title;
+		}
+		if ($state->photo->description === null) {
+			$state->photo->description = $state->exifInfo->description;
+		}
+		if (count($state->photo->tags) === 0) {
+			$state->photo->tags = $state->exifInfo->tags;
```

Comment:
Not immediately relevant: I wonder if we should merge tags if different. I don't think it's likely to come up often enough to be worth the effort though.